src/Domain/Runner.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\PhpInsights\Domain;
+
+use NunoMaduro\PhpInsights\Domain\Contracts\Repositories\FilesRepository;
+use NunoMaduro\PhpInsights\Domain\FileProcessors\SniffFileProcessor;
+use NunoMaduro\PhpInsights\Domain\Insights\SniffDecorator;
+use SplFileInfo;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Finder\SplFileInfo as SymfonySplFileInfo;
+
+/**
+ * @internal
+ */
+final class Runner
+{
+    /** @var \NunoMaduro\PhpInsights\Domain\FileProcessors\SniffFileProcessor */
+    private $phpCsFileProcessor;
+
+    /** @var \Symfony\Component\Console\Output\OutputInterface */
+    private $output;
+
+    /** @var \NunoMaduro\PhpInsights\Domain\Contracts\Repositories\FilesRepository */
+    private $filesRepository;
+
+    /**
+     * InsightContainer constructor.
+     *
+     * @param \Symfony\Component\Console\Output\OutputInterface $output
+     * @param \NunoMaduro\PhpInsights\Domain\Contracts\Repositories\FilesRepository $filesRepository
+     */
+    public function __construct(
+        OutputInterface $output,
+        FilesRepository $filesRepository
+    )
+    {
+        $this->filesRepository = $filesRepository;
+        $this->output = $output;
+
+        $container = Container::make();
+
+        $this->phpCsFileProcessor = $container->get(SniffFileProcessor::class);
```

Comment:
When multiple processors are added, we should maybe consider getting a repository of them instead, else we would have to do this for each processor.